### DISCUSSION TITLE
Questions about formulas

### DISCUSSION THREAD

**Discussion by mattiason** [Budibase/budibase#3109] [Help] [answered]
Hi i am trying to get my head around budibase formulas

I have 2 things i need to solve 

**Case 1:** I have a relationship control where I can select multiple item in 

![image](https://user-images.githubusercontent.com/45860668/138345913-d5550b0e-0397-484e-92cc-26545ef48ba1.png)

Now I would like to find a formula to create a string like PropulsionEngine/Cylinder out from those? Is this possible and then how? 

**Case 2:** I have 1 text field and 1 options control with yes and no. For example I have Engine in the text file and now I want a calculated field that makes the following. If the option control = Yes then I want to concatenate the text field with "_#" so the calculated formula e.g. become Engine_#.

Is this possible?

**Comment by mjashanks** [comment, marked as answer]
Hey there.

**Case 1**
I know exactly what you mean, and it's not possible. We do hope to address this very soon. It's actually related to this: https://github.com/Budibase/budibase/discussions/2617 - I've commented below it to link back to this issue - please give it an upvote!

**Case 2**
This is possible now, but will become a lot easier once we have javascript - in a few weeks!

For now, the calculate formula should be `{{#if (eq Form.Fields.YesNo "Yes")}} #{{/if}}{{Form.Fields.Text}}`

Broken down, this means
- If the Form Field "YesNo" equals "Yes" then add a "#"
- Add Form Field "Text"

Here is my app export, if you want to see it in action:

[handlebars-if-export.txt](https://github.com/Budibase/budibase/files/7395833/handlebars-if-export.txt)

Also, you can find more on this syntax here: https://handlebarsjs.com/guide/builtin-helpers.html#if
